### PR TITLE
miscellaneous updates to kitchen environment

### PR DESCRIPTION
**Pull request by tsilver-bdai**
changes from @wmcclinton 

working towards getting the following command to produce reasonable performance:

```
python predicators/main.py --env kitchen --approach active_sampler_learning --experiment_id kitchen-random-explore --debug --strips_learner oracle --sampler_learner oracle --bilevel_plan_without_sim True --max_initial_demos 0 --num_train_tasks 100 --num_test_tasks 10 --max_num_steps_interaction_request 50 --sampler_mlp_classifier_max_itr 100000 --mlp_classifier_balance_data False --pytorch_train_print_every 10000 --active_sampler_learning_use_teacher False --online_nsrt_learning_requests_per_cycle 10 --num_online_learning_cycles 5 --explorer random_nsrts --seed 0 --active_sampler_learning_explore_length_base 10000
```

at the moment, performance starts out at 30%, then shoots up to 100% after one cycle, then crashes to ~20-30% and stays there. need to investigate, but first want to merge these various fixes.

### CODE REVIEW

File: predicators/envs/kitchen.py [hunk 1]
```diff
@@ -146,7 +143,8 @@ def simulate(self, state: State, action: Action) -> State:
 
     def step(self, action: Action) -> Observation:
         self._gym_env.step(action.arr)
-        self._gym_env.render()
+        if self._using_gui:
```
Comment (wmcclinton) [inline predicators/envs/kitchen.py:146]: Did we not already have a flag for rendering in other environments?

File: predicators/ground_truth_models/kitchen/nsrts.py [hunk 2]
```diff
@@ -70,9 +69,11 @@ def push_sampler(state: State, goal: Set[GroundAtom],
             y = state.get(gripper, "y")
             z = state.get(gripper, "z")
             if obj.name == 'knob3':
-                return np.array([x + 0.25, y, z], dtype=np.float32)
+                return np.array([x + 1.0, y, z], dtype=np.float32)
             if obj.name == 'kettle':
-                return np.array([x, y + 5.0, z - 0.3], dtype=np.float32)
+                rand_dx = _rng.uniform(0.0, 1.0)
```
Comment (wmcclinton) [inline predicators/ground_truth_models/kitchen/nsrts.py:74]: We are only adding noise to the x parameter when pushing the kettle? I guess for now it is ok, but eventually learning on top of one more generic sampler is what we want.

File: predicators/ground_truth_models/kitchen/options.py [hunk 3]
```diff
@@ -94,16 +94,47 @@ def policy(state: State, memory: Dict, objects: Sequence[Object],
...
+                if "target_pose" not in memory:
+                    memory["target_pose"] = np.array(
+                        [params[0], params[1], params[2]])
+
+                if np.allclose(np.array([gx, gy, gz]),
+                               memory["target_pose"],
+                               atol=0.2):
+                    return True
+            else:
+                grounded_op = operator.ground(tuple(objects))
```
Comment (wmcclinton) [inline predicators/ground_truth_models/kitchen/options.py:129]: You or I should add comments here saying the during MoveTo we terminate based on getting to the initial sampled location and in every other option we terminate during add_effect completion.

File: predicators/main.py [hunk 4]
```diff
@@ -77,7 +77,8 @@ def main() -> None:
         handlers.append(logging.FileHandler(CFG.log_file, mode='w'))
     logging.basicConfig(level=CFG.loglevel,
                         format="%(message)s",
-                        handlers=handlers)
+                        handlers=handlers,
+                        force=True)
```
Comment (wmcclinton) [inline predicators/main.py:81]: nice!

**Review by wmcclinton — COMMENTED** [review]
Thanks so much!

File: predicators/envs/kitchen.py (hunk 1 above)
Comment (tsilver-bdai) [reply predicators/envs/kitchen.py:146]: yeah we do -- this is using existing functionality. we can now use `--use_gui` and that toggles `self._using_gui` in the base env class

File: predicators/ground_truth_models/kitchen/nsrts.py (hunk 2 above)
Comment (tsilver-bdai) [reply predicators/ground_truth_models/kitchen/nsrts.py:74]: yeah this was just enough to get it to like 30% base success

**Review by nkumar-bdai — APPROVED** [review]
LGTM!

**Pull request merged**